packages/phoenix-event-display/src/managers/three-manager/index.ts

```diff
@@ -238,23 +243,33 @@ export class ThreeManager {
    * @param openingAngle The opening angle of clipping.
    */
   public setClippingAngle(startingAngle: number, openingAngle: number) {
-    const startingAngleQuaternion = new Quaternion();
-    startingAngleQuaternion.setFromAxisAngle(
-      new Vector3(0, 0, 1),
-      (startingAngle * Math.PI) / 180
-    );
-    this.clipPlanes[0].normal
-      .set(0, -1, 0)
-      .applyQuaternion(startingAngleQuaternion);
-
-    const openingAngleQuaternion = new Quaternion();
-    openingAngleQuaternion.setFromAxisAngle(
-      new Vector3(0, 0, 1),
-      ((startingAngle + openingAngle) * Math.PI) / 180
-    );
-    this.clipPlanes[1].normal
-      .set(0, 1, 0)
-      .applyQuaternion(openingAngleQuaternion);
+      this.clipIntersection = (openingAngle < 180);
+      const startingAngleQuaternion = new Quaternion();
+      startingAngleQuaternion.setFromAxisAngle(
+          new Vector3(0, 0, 1),
+          (startingAngle * Math.PI) / 180
+      );
+      this.clipPlanes[0].normal
+          .set(0, -1, 0)
+          .applyQuaternion(startingAngleQuaternion);
+      const openingAngleQuaternion = new Quaternion();
+      openingAngleQuaternion.setFromAxisAngle(
+          new Vector3(0, 0, 1),
+          ((startingAngle + openingAngle) * Math.PI) / 180
+      );
+      this.clipPlanes[1].normal
+          .set(0, 1, 0)
+          .applyQuaternion(openingAngleQuaternion);
+      const geometries = this.sceneManager.getGeometries();
+      for (const geometry of geometries.children) {
```

Comment:
I think going through all the detector geometry objects each time we change the clipping is going to consume A LOT of computing power.

It will be better to have an explicit option to enable clipping intersection manually next to the clipping options.

Reply:
To my knowledge (which I admit is till limited), there is no other choice when you want to implement 360 degree clipping. The manual option cannot work as you need to change it automatically when hitting the 180 degree boundary.

Reply:
I've just tested it with LHCb full geometry and it seems like it's absolutely fine

Reply:
You are correct, it does seem to work fine. One quick comment is that it doesn't seem to work for LHCb although it works fine for ATLAS. The geometry disappears for me for values of opening angle greater than 180 degrees.

Reply:
The problem with the (current) LHCb is  what I explain in the description of the MR : collision of this feature with user defined clipping plane (here the z = 20000 one). In such a case, switching to intersection removes everything indeed.
This is why I'm wondering if we should not finally remove this feature that we've just introduced. It's not used by anyone (but LHCb, but by mistake, I will drop it in a MR tomorrow) and collides with the 360 degree clipping.

What do you think ?

Reply:
Ah, I get it now. Thanks for explaining it again. I missed it in the description.

By feature, if you mean defining extra clipping planes through configuration, then I don't have a strong opinion either way. Whatever suits your use case should work.

Reply:
Ok, then let's drop these extra planes for now as noone needs them, and we'll come back to this in case.
I'm preparing a new version of this MR